### ISSUE TITLE
refactor: Migrate Bayes + Similarity checks to unified ML.NET text classifier with TF-IDF

### ISSUE THREAD

**Issue by musicislife08** [musicislife08/TelegramGroupsAdmin#159]
## Problem

Two content detection checks need improvement:

1. **SimilarityContentCheckV2** - Uses simplified Jaccard instead of proper TF-IDF (line 95 comment acknowledges this)
2. **BayesContentCheckV2** - Hand-rolled Naive Bayes that retrains hourly from scratch

Both checks are trying to answer the same fundamental question: *"Does this text look like known spam?"*

## Solution: Unified ML.NET Text Classifier

Since ML.NET is already installed (`Microsoft.ML 5.0.0`) and used successfully for threshold optimization (`ThresholdRecommendationService`), migrate both checks to a single ML.NET text classification model.

### Benefits

- **Built-in TF-IDF** - ML.NET TextFeaturizer includes TF-IDF, n-grams, stop word removal
- **Better algorithms** - SdcaLogisticRegression typically outperforms hand-rolled Naive Bayes
- **Model persistence** - Save trained model to disk, skip cold-start retraining
- **Unified check** - One model replacing two, simpler architecture
- **Consistent patterns** - Matches existing ThresholdOptimizationFeatures approach

### Algorithm Choice: SDCA Logistic Regression

Start with `SdcaLogisticRegression` (same as ThresholdRecommendationService) rather than LightGBM:
- Faster to train on small/medium datasets
- More stable with class imbalance
- Already proven in codebase
- Keep trainer swappable for future experimentation

### Training Data

Include both manual and auto-detected labels:
- **Manual corrections** (`/spam`, UI marking) - highest quality signal
- **Auto-detected** - GPT-validated, accurate enough to include

Use translated text (already pipeline standard).

### Training Pipeline

```csharp
var pipeline = mlContext.Transforms.Text
    .FeaturizeText("TextFeatures", nameof(SpamTextFeatures.MessageText))  // TF-IDF + n-grams
    .Append(mlContext.Transforms.Concatenate("Features",
        "TextFeatures",
        nameof(SpamTextFeatures.MessageLength),
        nameof(SpamTextFeatures.HasUrls)))
    .Append(mlContext.BinaryClassification.Trainers.SdcaLogisticRegression(
        labelColumnName: "Label",
        featureColumnName: "Features"));
```

### Model Persistence

```
/data/ml-models/
├── spam-classifier.zip           # Current active model
├── spam-classifier.meta.json     # Training metadata (sample count, date)
└── spam-classifier.backup.zip    # Previous version for rollback
```

## Intelligent Retraining Strategy

| Trigger | When | Rationale |
|---------|------|-----------|
| **Startup** | Always | Load persisted model or train if missing |
| **Scheduled** | Configurable (default: 4 hours?) | Regular refresh for gradual drift |
| **False Negative Detection** | After N admin corrections via /spam or UI | Catch spam waves immediately |

### Event-Driven Retraining

When admins manually mark messages as spam that the pipeline missed:

```
Message arrives → Pipeline says "ham" → Gets through
        ↓
Admin uses /spam command OR marks spam in UI
        ↓
This was a FALSE NEGATIVE (high-value training signal)
        ↓
Increment counter
        ↓
If counter >= threshold (e.g., 5 false negatives)
        ↓
Trigger model retraining job
```

Note: No complex guardrails needed - admin corrections are infrequent (1-2/week typically), and the trust network would catch abuse quickly.

## Rollout Strategy: Shadow Mode

Instead of a feature flag, introduce the new classifier in **shadow mode**:

```csharp
// In TextClassifierContentCheckV2.CheckAsync()
var prediction = await _classifierService.PredictAsync(message);

// Shadow mode: Always abstain, but show what would have happened
return new ContentCheckResponseV2
{
    CheckName = CheckName.TextClassifier,
    Score = 0.0,  // Doesn't affect total
    Abstained = true,
    Details = $"Shadow mode: Would have scored {prediction.Score:F1} ({prediction.Probability:P0} confidence)"
};
```

Benefits:
- **Zero-risk rollout** - Cannot cause false positives/negatives
- **Real production comparison** - See exactly how it would have performed
- **Appears in detection results UI** - Observable without affecting decisions
- **Easy promotion** - Remove abstain override when confident

### Phase 1: Create New ML.NET Check (Shadow Mode)
- [ ] Create MLTextClassifierService with training and prediction
- [ ] Create SpamTextFeatures and SpamPrediction classes
- [ ] Implement model persistence (save/load from /data/ml-models/)
- [ ] Create TextClassifierContentCheckV2 implementing IContentCheckV2
- [ ] Add to pipeline in shadow mode (abstain + show prediction)

### Phase 2: Retraining Infrastructure
- [ ] Add ModelRetrainingJob Quartz job
- [ ] Implement false-negative trigger (counter after N admin corrections)
- [ ] Hook into /spam command and UI spam marking flows
- [ ] Add startup model loading with fallback to fresh training
- [ ] Add atomic model swap (Interlocked reference replacement)

### Phase 3: Promote from Shadow Mode
- [ ] Compare shadow predictions against actual outcomes
- [ ] Remove shadow mode override when accuracy is confirmed
- [ ] Update config UI if needed

### Phase 4: Cleanup
- [ ] Deprecate BayesContentCheckV2 and SimilarityContentCheckV2
- [ ] Remove old BayesClassifier class
- [ ] Remove simplified Jaccard similarity code
- [ ] Update documentation

## Configuration Options

```csharp
public class TextClassifierConfig
{
    public bool Enabled { get; set; } = true;
    public bool ShadowMode { get; set; } = true;  // Start in shadow mode
    public int RetrainAfterFalseNegatives { get; set; } = 5;
    public string ScheduledRetrainCron { get; set; } = "0 0 */4 * * ?";  // Every 4 hours
    public int MinTrainingSamples { get; set; } = 50;
    public double SpamProbabilityThreshold { get; set; } = 0.7;
}
```

## Scoring Integration

Map model probability to additive score (0-5 points) to fit existing pipeline:

| Probability | Score | Rationale |
|-------------|-------|-----------|
| >= 0.95 | 5.0 | Very high confidence spam |
| >= 0.85 | 3.5 | High confidence |
| >= 0.70 | 2.0 | Moderate confidence |
| >= 0.60 | 1.0 | Low confidence |
| < 0.60 | 0.0 (abstain) | Too uncertain |

## Success Metrics

- [ ] Model trains successfully on existing training data
- [ ] Shadow mode predictions appear in detection results UI
- [ ] False-negative retraining triggers correctly
- [ ] Model persists across restarts
- [ ] Cold start time acceptable (< 30 seconds)
- [ ] Shadow mode shows >= current Bayes accuracy before promotion

## References

- Existing ML.NET usage: ThresholdRecommendationService.cs
- ML.NET Text Classification: https://learn.microsoft.com/en-us/dotnet/machine-learning/tutorials/sentiment-analysis
